packages/titus-backend/lib/plugins/auth/authzcheck.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const { Forbidden } = require('http-errors')
+const jwt = require('jsonwebtoken')
+
+async function authRoutes(server, options) {
+  // add users to the admin role
+  server.addHook('onReady', async function () {
+    const adminUsers = process.env.CHECK_AUTHZ_ADMIN_USERS || ''
+    const policies = adminUsers
+      .split(',')
+      .map(u => server.casbin.addRoleForUser(u, 'role_admin'))
+    await Promise.all(policies)
+  })
```

Comment:
Why did you put this in `onReady` and not in line? If you need to specifify a plugin dependency with fastify-autoload you can use the plugin dependencies: https://github.com/fastify/fastify-autoload#plugin-configuration:

```js
// plugins/plugin-a.js
const fp = require('fastify-plugin')

function plugin (fastify, opts, next) {
  // plugin a
}

module.exports = fp(plugin, {
  name: 'plugin-a',
  dependencies: ['plugin-b']
})

// plugins/plugin-b.js
function plugin (fastify, opts, next) {
  // plugin b
}

module.exports = fp(plugin, {
  name: 'plugin-b'
})
```

Reply:
yes, defining the dependencies is much better. - Thanks